### DISCUSSION TITLE
Blog comment on Multi-Tenancy Vector Search with millions of tenants

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#916] [Docs/blog comments]
# Blog comment on Multi-Tenancy Vector Search with millions of tenants

Learn about the new native multi-tenancy feature

https://weaviate.io/blog/multi-tenancy-vector-search

**Comment by roengram** [comment]
Great feature! I just have one question: why the workaround of using classes can support merely 5 ~ 10,000 tenants? Is it the maximum number of classes one can create in Weaviate?

**Comment by kmohanarangam** [comment]
Thanks for this article. Awesome innovation on natively supporting multi tenancy. This feature isolates tenant's data. Are there any tools that will help with access control (similar to row level or column level access control in RDBMS).?  For example, the application layer is going to be common for all tenants but the data will be isolated. How can the application access be restricted to certain tenant's data alone.

**Comment by FalconDinesh** [comment]
Can we able to project our queries to two or more tenants like .with_tenant("tenant-2021*"), such that we can query on matching tenant names, basically same as using LIKE in SQL. Is the feature available in Weaviate?

**Comment by spyrosigma271** [comment]
Multi-Tenancy is great !!  
Would love if it was also a part of Unstructured ETL pipelines. 

I was trying to get Make ETL pipelines from aws-S3 to Weaviate for multiple users, But couldn't find a way to utilize Weaviate's Multi-Tenancy via Unstructured. 

Now, Need to create separate Collection for Each User !


Is there any alternative for this?
